frontend/src/metabase/collections/components/BulkActions.tsx

```
@@ -79,45 +156,96 @@ const BulkActions = ({
                 )}
               </CardSide>
               <CardSide>
-                <CardButton
-                  medium
-                  purple
-                  disabled={!canMove}
-                  onClick={onMoveStart}
-                >{t`Move`}</CardButton>
-                <CardButton
-                  medium
-                  purple
-                  disabled={!canArchive}
-                  onClick={() => {
-                    onArchive?.();
-                  }}
-                >{t`Archive`}</CardButton>
+                {isTrashedCollection(collection) ? (
+                  <>
+                    {showUnarchive && (
+                      <CardButton
+                        medium
+                        purple
+                        onClick={() => handleBulkSetArchive(false)}
+                        disabled={!canUnarchive}
+                      >
+                        {t`Restore`}
+                      </CardButton>
+                    )}
+                    <CardButton
+                      medium
+                      purple
+                      onClick={handleBulkMoveStart}
```

the silently passed event argument catches my eye but probably no biggie